frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_property_edit/mobile_field_type_option_editor.dart

```diff
@@ -0,0 +1,197 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:go_router/go_router.dart';
+
+import 'mobile_field_type_list.dart';
+
+class MobileFieldTypeOptionEditor extends StatelessWidget {
+  final TypeOptionController _dataController;
+
+  const MobileFieldTypeOptionEditor({
+    required TypeOptionController dataController,
+    Key? key,
+  })  : _dataController = dataController,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) {
+        final bloc = FieldTypeOptionEditBloc(_dataController);
+        bloc.add(const FieldTypeOptionEditEvent.initial());
+        return bloc;
+      },
+      child: BlocBuilder<FieldTypeOptionEditBloc, FieldTypeOptionEditState>(
+        builder: (context, state) {
+          final typeOptionWidget = _makeTypeOptionMobileWidget(
+            context: context,
+            dataController: _dataController,
+          );
+
+          return Column(
+            children: [
+              const _MobileSwitchFieldButton(),
+              const VSpace(8),
+              if (typeOptionWidget != null) typeOptionWidget
+            ],
+          );
+        },
+      ),
+    );
+  }
+}
+
+class _MobileSwitchFieldButton extends StatelessWidget {
+  const _MobileSwitchFieldButton({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final fieldType = context.select(
+      (FieldTypeOptionEditBloc bloc) => bloc.state.field.fieldType,
+    );
+    return GestureDetector(
+      child: Row(
+        children: [
+          Text(
+            LocaleKeys.grid_field_propertyType.tr(),
+            style: Theme.of(context).textTheme.titleMedium,
+          ),
+          const Spacer(),
+          FlowySvg(fieldType.icon()),
+          const HSpace(4),
+          Text(
+            fieldType.title(),
+            style: Theme.of(context).textTheme.titleMedium,
+          ),
+          Icon(
+            Icons.arrow_forward_ios_sharp,
+            color: Theme.of(context).hintColor,
+          ),
+        ],
+      ),
+      onTap: () => showFlowyMobileBottomSheet(
+        context,
+        isScrollControlled: true,
+        title: LocaleKeys.grid_field_propertyType.tr(),
+        builder: (_) => MobileFieldTypeList(
+          bloc: context.read<FieldTypeOptionEditBloc>(),
+          onSelectField: (newFieldType) {
+            context.read<FieldTypeOptionEditBloc>().add(
+                  FieldTypeOptionEditEvent.switchToField(newFieldType),
+                );
+            context.pop();
+          },
+        ),
+      ),
+    );
+  }
+}
+
+Widget? _makeTypeOptionMobileWidget({
+  required BuildContext context,
+  required TypeOptionController dataController,
+}) {
+  final builder = _makeTypeOptionMobileWidgetBuilder(
+    dataController: dataController,
+  );
+  return builder.build(context);
+}
```

Comment:
Personal preference, formatting is off in the suggestion
```suggestion
}) =>  
  _makeTypeOptionMobileWidgetBuilder(
    dataController: dataController,
  ).build(context);
```